extract.py

```diff
@@ -10,16 +10,19 @@
         'application/zip',
         'application/x-zip-compressed',
         'application/octet-stream',
+        'application/xml',
         ]
 
 def save_file(attachment, mtime):
     f_name = attachment.get_filename()
-    dest = os.path.join('./reports', f_name)
-    if os.path.exists(dest):
-        return
-    with open(dest, 'wb') as f:
-        f.write(attachment.get_payload(decode=True))
-    os.utime(dest, (mtime, mtime))
+
+    if f_name:
+        dest = os.path.join('./reports', f_name)
+        if os.path.exists(dest):
+            return
+        with open(dest, 'wb') as f:
+            f.write(attachment.get_payload(decode=True))
+        os.utime(dest, (mtime, mtime))
```

Comment:
If the 'f_name' variable is undifined, I assume that the mail does not have an attachment.